tests/virt_autotest/qem/live_migration_src.pm

```diff
@@ -0,0 +1,213 @@
+        $profile =~ s/\{\{VERSION\}\}/$vm_info->{version}/g;
+        $profile =~ s/\{\{SUT_IP\}\}/$sut_ip/g;
+        $profile =~ s/\{\{GUEST\}\}/$vm_name/g;
+        $profile =~ s/\{\{DISK_TYPE\}\}/$vm_info->{disk_type}/g;
+        $profile =~ s/\{\{IP_PATH\}\}/$nfs_path\/guestip/g;
+        save_tmp_file("$vm_name.xml", $profile);
+        $vm_info->{autoyast} = autoinst_url . "/files/$vm_name.xml";
+        record_info($vm_info->{autoyast});
+        # Copy profile to ulogs directory, so profile is available in job logs
+        make_path('ulogs');
+        copy(hashed_string("$vm_name.xml"), "ulogs/$vm_name.xml");
+        $vm_info->{install_cmd} = "$virt_insatll --name $vm_name $vm_info->{v_type} --disk path=$vm_info->{path} --location $vm_info->{location} -x autoyast=$vm_info->{autoyast} --graphics vnc,port=$vm_info->{port},listen=0.0.0.0";
+        record_info($vm_info->{install_cmd});
+    }
+}
+
+
+sub run {
+    select_console 'root-ssh';
+    $dst_ip   = get_job_autoinst_vars(get_parents->[0])->{SUT_IP};
+    $inst_src = get_test_suite_data()->{location}->{"$guest"};
+    record_info("DST: $dst_ip");
+    record_info("SUT: $sut_ip");
+    assert_script_run("if [ -d $images_path/guestip ]; then rm -rf $images_path/guestip; fi; mkdir $images_path/guestip");
+    assert_script_run("if ! [ -d $images_path/vm-configs ]; then mkdir $images_path/vm-configs; fi");
+    init_vm_info;
+    #start vm installation
+    if ($vms_to_install > 0) {
+        init_vm_profile;
+        while (my ($vm_name, $vm_info) = each %$vms) {
+            next if $vm_info->{reuse};
+            assert_script_run($vm_info->{install_cmd});
+        }
+        sleep 600;
+    }
+    #boot vms into second installation stage and verify install is done
+    my $n         = keys %$vms;
+    my $time_cont = 0;
+    until ($n == 0) {
+        sleep 120;
+        $time_cont++;
+        while (my ($vm_name, $vm_info) = each %$vms) {
+            unless ($vm_info->{ip} =~ /^\d+\.\d+\.\d+\.\d+$/) {
+                $vm_info->{ip} = script_output("if [ -f $images_path/guestip/$vm_name ]; then cat $images_path/guestip/$vm_name; fi");
+                if ($vm_info->{ip} =~ /^\d+\.\d+\.\d+\.\d+$/) {
+                    $n--;
+                } else {
+                    die("$vm_name startup time out after " . $time_cont * 2 . " minutes") if ($time_cont == 5);
+                    my $vm_state = script_output("virsh domstate --domain $vm_name");
+                    assert_script_run("virsh start $vm_name") if ($vm_state eq "shut off");
+                }
+            }
+            record_info($vm_info->{ip});
+        }
+    }
+
+    mutex_wait 'dst_ready';
+    # set up ssh login dst host without password
+    assert_script_run('if [ -d /root/.ssh ]; then rm -rf /root/.ssh; fi');
+    assert_script_run('ssh-keygen -t rsa -q -f "/root/.ssh/id_rsa" -N ""');
+    assert_script_run("ssh-keyscan $dst_ip > /root/.ssh/known_hosts");
```

Comment:
We do all of this in `virtualization/universal/ssh_hypervisor_init.pm` test module.

Reply:
ssh_hypervisor_init will run some code which is not needed and relevant in this test. I created this key just for login dst host without password. I also don't want sshpass which need add packagehub repo. In product QA senorio packagehub might get ready a few months later than other repo.   

The varibles defined in the begining of code are not only used by guest install but by guest migration and vm backup. Spliting up code will make it hard to transfer those data. maybe use set_var but complicate the test very much.

Reply:
This modules are ready - just reuse them instead of writing new ones.
That modul does not use sshpass. If you don't need some part of that module just add condition.

Reply:
It does not use sshpass but exec_and_insert_password is way more complicated: it's using needle! How many code does it run to copy pub key?  Mine is just curl command. 
You know that  the less code it runs the more robust it will be.

Reply:
It is universal, used by everywhere. 

DOn't get me wrong - I like your solution - I'd prefer it. But it is different topic.

You may create another pull request (or at least another commit) and improve the `ssh_hypervisor_init` test module.